lfsapi/endpoint_finder.go

```diff
@@ -202,39 +216,38 @@ func (e *endpointGitFinder) NewEndpoint(rawurl string) lfshttp.Endpoint {
 	}
 }
 
-func (e *endpointGitFinder) AccessFor(rawurl string) AccessMode {
+func (e *endpointGitFinder) AccessFor(rawurl string) Access {
+	accessurl := urlWithoutAuth(rawurl)
+
 	if e.gitEnv == nil {
-		return NoneAccess
+		return Access{mode: NoneAccess, url: accessurl}
 	}
 
-	accessurl := urlWithoutAuth(rawurl)
-
 	e.accessMu.Lock()
 	defer e.accessMu.Unlock()
 
 	if cached, ok := e.urlAccess[accessurl]; ok {
-		return cached
+		return Access{mode: cached, url: accessurl}
 	}
 
 	e.urlAccess[accessurl] = e.fetchGitAccess(accessurl)
-	return e.urlAccess[accessurl]
+	return Access{mode: e.urlAccess[accessurl], url: accessurl}
 }
 
-func (e *endpointGitFinder) SetAccess(rawurl string, access AccessMode) {
-	accessurl := urlWithoutAuth(rawurl)
-	key := fmt.Sprintf("lfs.%s.access", accessurl)
-	tracerx.Printf("setting repository access to %s", access)
+func (e *endpointGitFinder) SetAccess(access Access) {
+	key := fmt.Sprintf("lfs.%s.access", access.url)
+	tracerx.Printf("setting repository access to %s", access.mode)
 
 	e.accessMu.Lock()
 	defer e.accessMu.Unlock()
 
-	switch access {
+	switch access.mode {
```

Comment:
What do you think about calling the `GetMode()`/`Mode()` function here instead? I think that that's a good way to go since it will minimize the diff if we ever need to do extra computation to determine the access mode.

This same comment applies throughout the rest of this file, and its test.